src/main/java/com/google/api/codegen/config/FieldConfig.java

```diff
@@ -210,6 +211,7 @@ public FieldConfig withResourceNameConfig(ResourceNameConfig resourceNameConfig)
   public boolean hasDifferentMessageResourceNameConfig() {
     return getResourceNameConfig() != null
         && getMessageResourceNameConfig() != null
+        && getMessageResourceNameConfig().getResourceNameType() != ResourceNameType.ANY
```

Comment:
I don't understand why this is necessary. Doesn't this mean that the `hasDifferentMessageResourceNameConfig()` function will return false whenever the message resource name is Any? That is incorrect if the method resource name is some more specific resource name or oneof.

Edit: looking into it further, I see that this function is being used to determine whether we should apply a parameter transformation, and that in the ANY case we don't want to do that transformation. Given that, can we either rename this method so that it describes itself accurately, or else move the check for ANY into the places this function is called instead.